packages/sirius-web/frontend/sirius-web-application/src/views/project-settings/ProjectSettingsView.tsx

```diff
@@ -123,12 +115,39 @@ export const ProjectSettingsView = () => {
   }, [error]);
   const project: GQLProject | null = data?.viewer.project;
 
-  const handleTabChange = (_event, newValue: string) =>
-    setState((prevState) => ({ ...prevState, selectedTabId: newValue }));
+  const selectedTabId: string | null =
+    tabId && projectSettingsTabContributions.find((tab) => tab.id == tabId)
+      ? tabId
+      : projectSettingsTabContributions[0]?.id ?? null;
+  const settingContentContribution: ProjectSettingTabContribution | null = selectedTabId
+    ? projectSettingsTabContributions.filter((contribution) => contribution.id === selectedTabId)[0]
+    : null;
+
+  const navigate = useNavigate();
+  const [urlSearchParams] = useSearchParams();
+  const handleTabChange = (_event, newValue: string) => {
+    let pathname: string | null = null;
+    if (newValue && newValue !== tabId) {
+      pathname = generatePath('/projects/:projectId/settings/:tabId', {
+        projectId,
+        tabId: newValue,
+      });
+    } else if (newValue === null && tabId) {
+      pathname = generatePath('/projects/:projectId/settings/', { projectId });
+    }
```

Comment:
How could the `newValue` (which has the type `string` and not `string | null` or `string | undefined` could be anything else than a string? I don't understand how we could end up with `/projects/:projectId/settings`. Why do we have anything more than this?

```
const handleTabChange = (_event, tabId: string) => {
  const pathname: generatePath('/projects/:projectId/settings/:tabId', {
    projectId,
    tabId,
  });
  navigate(pathname);
};
```

Why all this defensive code?